vendor/src/github.com/docker/swarmkit/ca/server.go

```diff
@@ -233,8 +233,9 @@ func (s *Server) IssueNodeCertificate(ctx context.Context, request *api.IssueNod
 					},
 				},
 				Spec: api.NodeSpec{
-					Role:       role,
-					Membership: api.NodeMembershipAccepted,
+					Role:         role,
+					Membership:   api.NodeMembershipAccepted,
+					Availability: request.Availability,
```

Comment:
Sorry, for some reason I thought this was a string. In this case, I think it makes sense to make it a pointer, and keep the value `nil` if there the user did not override it. On the gRPC side, this means splitting it out into a separate message.